### ISSUE TITLE
Setting a flag for a required operation resource casts wrong if not alphanumeric

### ISSUE THREAD

**Issue by ghwatson** [guildai/guildai#259]
Context: I recently created an evaluation operation which has a train operation as a resource. Poking through the docs, I understood that I can then use `train` as one of my flags. Taking advantage of the ability to just partially type in a run id, I put in the first 5 characters which included no letters. So I had  `guild run ..... train=05930`. 

Bug: Guild interpreted my input flag as a number rather than a run id, and lists the value of the flag as the number '5930' as opposed to a full run id.

Easy to work around this by just including a few more run id characters until it becomes alphanumeric, then it interprets it correctly and lists the full run id at confirmation time before kicking off the run. Just wanted to submit this in case this is undesired behaviour. I figure given the guild.yml context, guild should be able to infer a run id is required and not a number.

**Comment by gar1t** [comment]
What version of Guild are you using? `guild check` gives you this.

I can't recreate this. Guild's behavior is to treat that value as a string and use it to resolve an upstream run. The fully resolved run ID should appear in the preview prompt when you run the downstream run. That's what I'm seeing on `0.7.2`.

**Comment by ghwatson** [comment]
Yep guild check gives me 0.7.2. Let me try to provide a bit more context (scrubbed some filenames) to see if that helps. Also, an extra clue for the bug and the reason I don't think you reproduced it at the end:

Here's my guild.yml:

```
train:
  description: Training script
  main: train_model
  flags-dest: globals
  flags-import: all

eval_1:
  description: Some eval routine.
  main: eval_1
  requires:
    - operation: train
  flags-dest: globals
  flags-import: all
  output-scalars: '^(\key): (\value)'
```

I have a prior train operation with run id = 05930ca30d0a407aadf166d7836cb688.

If I do `guild run eval_1 train=05930` I get this returned:

```
WARNING: cannot find a suitable run for required resource 'train'
You are about to run eval_1
  DEVMODE: yes
  percentile_cutoff: default
  plane_reduce: middle
  train: '5930'
Continue? (Y/n) Y
Resolving train dependency
guild: run failed because a dependency was not met: could not resolve 'operation:train' in train resource: no suitable run for train
```

If I do `guild run eval_1 train=05930c` to make it not just purely numbers I get:

```
You are about to run eval_1
  DEVMODE: yes
  percentile_cutoff: default
  plane_reduce: middle
  train: 05930ca30d0a407aadf166d7836cb688
Continue? (Y/n)
```

which worked fine. 

To further test what was going on, I then tried with another train run id where I could just utilize a part of the hash that is entirely numeric (motivated by how the 0 gets cut out of the train flag in the bugged case's flag confirmation above). It seems you need a run id that starts with 0. Another run id that started with 584427 did not have this issue. Perhaps this is why you weren't able to reproduce it given my initial description.

TLDR: bug seems to only occur for partial run ids that are purely numeric _**and**_ begin with "0".

**Comment by gar1t** [comment]
Sorry for the no reply here. We finished this up back in Feb but I forgot to update you.

[This doc](https://github.com/guildai/issue-resolution/blob/master/259-setting-a-flag-for-a-required-operation-resource-casts-wrong-if-not-alphanumeric/FIX.md) shows the correct behavior, which I confirmed is working for the current pre-release of 0.7.3. The production release should be going out today or tomorrow.

Thanks again for the detailed report. Very helpful!
